tic_tac_toe/game.py

```diff
@@ -87,18 +147,53 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game["board"]
+    nextTurn = game["next_turn"]
+    playerOne = game["player1"]
+    playerTwo = game["player2"]
+    weHaveWinner = game["winner"]
+    
+    if (_board_is_full(board) or weHaveWinner):
+        raise InvalidMovement("Game is over.")
+    elif not (_position_is_valid(position)):
+        raise InvalidMovement("Position out of range.") 
+    elif not (_position_is_empty_in_board(position, board)):
+        raise InvalidMovement("Position already taken.")
+    elif (nextTurn != player):
+        raise InvalidMovement("\"{}\" moves next".format(nextTurn))
```

Comment:
You could also say this:

`raise InvalidMovement('"{}" moves next'.format(nextTurn))`